tests/functional/spec/timeout_modal.spec.js

```diff
@@ -0,0 +1,70 @@
+import TimeoutInterstitialPage from "../generated_pages/timeout_modal/timeout-modal-interstitial.page";
+import TimeoutSubmitPage from "../generated_pages/timeout_modal/submit.page";
+import { TimeoutModalPage } from "../base_pages/timeout-modal.page.js";
+import ThankYouPage from "../base_pages/thank-you.page.js";
+
+describe("Timeout Modal", () => {
+  describe("Given I am completing the survey, ", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("test_timeout_modal.json");
+    });
+    testCase(TimeoutInterstitialPage);
+  });
+});
+
+describe("Timeout Modal Post Submission", () => {
+  describe("Given I am completing the survey and get to post submission page, ", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("test_timeout_modal.json");
+      $(TimeoutInterstitialPage.submit()).click();
+      $(TimeoutSubmitPage.submit()).click();
+    });
+    testCase(ThankYouPage);
+  });
+});
+
+function testCase(page) {
+  it("When the timeout modal is displayed, and I do not extend my session, Then I will be redirected to the session expired page", () => {
+    checkTimeoutModal();
+    browser.pause(65000); // We are waiting for the session to expire
+    expect(browser.getUrl()).to.contain("/session-expired");
+    expect($("body").getHTML())
+      .to.include(
+        "Sorry, you need to sign in again",
+        "This is because you have either:",
+        "been inactive for 45 minutes and your session has timed out to protect your information",
+        "followed a link to a page you are not signed in to",
+        "followed a link to a survey that has already been submitted"
+      )
+      .to.not.include("To protect your information, your progress will be saved and you will be signed out in");
+  });
+
+  it("When the timeout modal is displayed, and I click the “Continue survey” button, Then my session will be extended", () => {
+    checkTimeoutModal();
+    $(TimeoutModalPage.submit()).click();
+    expect($(TimeoutModalPage.timer()).getText()).to.equal("");
+    browser.pause(15000); // Waiting 15 seconds to sanity check it hasn't redirected
```

Comment:
With this pause and the one below, the pause has to be long enough to where the session would have expired. i.e timeout modal pops up at 60 seconds, if we then wait 15 second, we still have 45 seconds until the session times out hence we cannot know it extended. So we need to wait ~65 seconds, then refresh.

Reply:
Extended this to 65 seconds now, luckily doesn't time out (our wdio.conf setting is 135 seconds).